crates/lsp/src/main_loop.rs

```diff
@@ -378,16 +378,9 @@ impl AuxiliaryState {
         // Set global instance of this channel. This is used for interacting
         // with the auxiliary loop (logging messages or spawning a task) from
         // free functions.
-        unsafe {
-            #[allow(static_mut_refs)]
-            if let Some(val) = AUXILIARY_EVENT_TX.get_mut() {
-                // Reset channel if already set. Happens e.g. on reconnection after a refresh.
-                *val = auxiliary_event_tx;
-            } else {
-                // Set channel for the first time
-                AUXILIARY_EVENT_TX.set(auxiliary_event_tx).unwrap();
-            }
-        }
+        AUXILIARY_EVENT_TX
+            .set(auxiliary_event_tx)
+            .expect("Auxiliary event channel can't be set more than once.");
```

Comment:
I had the very important realization that unlike in the current ark setup, we won't ever go through here twice in air. There is no "reconnect" lifecycle to manage, our whole process is just going to get dropped and reestablished, so we don't have to worry about this anymore.

This _was_ being triggered in our tests though - even if we ran the tests sequentially we'd come through here multiple times as we "start up" a new LSP within the same process. I've fixed that by running the LSP tests as integration tests with 1 client per process, truly ensuring we never run through here again.